### PR TITLE
docs(treeview): update the rename node kb

### PR DESCRIPTION
**Pull request by YanushevGeorgi**

### CODE REVIEW

File: knowledge-base/treeview-rename-node.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ---
 title: How to Rename a TreeView Node
-description: How to edit the name (text) of a treeview node
+description: Learn how to edit the name (text) of a treeview node and explore the example that demonstrates a possible approach with an ItemTemplate.
```
Comment (yordan-mitev) [inline knowledge-base/treeview-rename-node.md:3]:
```suggestion
description: Learn how to edit the name (text) of a TreeView node and explore the example that demonstrates a possible approach with an ItemTemplate. 
```

File: knowledge-base/treeview-rename-node.md [hunk 2]
```diff
@@ -22,11 +22,184 @@ res_type: kb
 
 
 ## Description
-Is there any way I can edit the node text within a Blazor TreeView? 
+This KB article answers the following question:
+ - Is there any way I can edit the node text within a Blazor TreeView?
```
Comment (yordan-mitev) [inline knowledge-base/treeview-rename-node.md:26]:
```suggestion
 - Is there any way I can edit the node text within a Blazor TreeView? 
 - How can I let the user rename a node?
```

File: knowledge-base/treeview-rename-node.md [hunk 3]
```diff
@@ -22,11 +22,184 @@ res_type: kb
 
 
 ## Description
-Is there any way I can edit the node text within a Blazor TreeView? 
+This KB article answers the following question:
```
Comment (yordan-mitev) [inline knowledge-base/treeview-rename-node.md:25]:
```suggestion
This KB article answers the following questions:
```

File: knowledge-base/treeview-rename-node.md [hunk 4]
```diff
@@ -22,11 +22,184 @@ res_type: kb
...
+ - Is there any way I can edit the node text within a Blazor TreeView? 
 
-I want to let the user rename a node.
 
 ## Solution
-You can use the `ItemTemplate` to define how the node text will render and the desired UI and UX for editing. Here is a sample project that demonstrates a possible implementation. It uses a custom component `EditableTreeNode` that renders buttons and a textbox to update the TreeView item.
+Use the `ItemTemplate` to define the desired UI and UX for editing and how the node text will render. For example a button to initiate editing, a TextBox to modify the name of the node, a button to save the changes.
```
Comment (yordan-mitev) [inline knowledge-base/treeview-rename-node.md:30]:
```suggestion
Use the `ItemTemplate` to determine the node text's rendering and customize the editing UI. For example, you can define a button that initiates editing, a TextBox that modifies the node's name, and another button that saves the changes.
```

File: knowledge-base/treeview-rename-node.md [hunk 5]
```diff
@@ -22,11 +22,184 @@ res_type: kb
...
+    {
+        public int Id { get; set; }
+        public string Text { get; set; }
+        public int? ParentId { get; set; }
+        public bool HasChildren { get; set; }
+        public ISvgIcon Icon { get; set; }
+        public bool Expanded { get; set; }
+    }
+}
+````
```
Comment (yordan-mitev) [inline knowledge-base/treeview-rename-node.md:205]: Add a See Also section with up to three related links.

**Review by dimodi — APPROVED** [review]

File: knowledge-base/treeview-rename-node.md [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-description: How to edit the name (text) of a treeview node
+description: Learn how to edit the name (text) of a Treeview node and explore the example that demonstrates a possible approach with an ItemTemplate. 
 type: how-to
-page_title: How to rename a TreeView node?
+page_title: How to Rename a TreeView Node?
 slug: treeview-kb-rename-node
 position: 
-tags: edit, rename, tree, treeview, note, item
-ticketid: 1525532
+tags: edit, rename, tree, treeview, node, item
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:8]: Tags should make sense outside of context. "Node" and "item" are too general. But you can use template if you like.

```suggestion
tags: telerik, treeview, tree, edit, template
```

File: knowledge-base/treeview-rename-node.md [hunk 7]
```diff
@@ -22,11 +22,190 @@ res_type: kb
...
+                }
+            </ItemTemplate>
+        </TreeViewBinding>
+    </TreeViewBindings>
+</TelerikTreeView>
+
+@code {
+    private TelerikTextBox TextBoxRef { get; set; }
+
+    private List<TreeItem> FlatData { get; set; }
```
Comment (dimodi) [inline knowledge-base/treeview-rename-node.md:67]: We usually put the data collection first.

```suggestion
    private List<TreeItem> FlatData { get; set; }

    private TelerikTextBox TextBoxRef { get; set; }
```

**Pull request merged**